text/0171-cloudfront-redesign.md

```diff
@@ -0,0 +1,495 @@
+    forwardQueryString: true,
+    edgeFunctions: [{
+      function: myFunctionVersion,
+      eventType: EventType.ORIGIN_RESPONSE,
+    }],
+  }),
+});
+dist.addOrigin(cloudfront.Origin.fromBucket(sourceBucket), { pathPattern: 'static/*' });
+```
+
+# Drawbacks
+
+The primary drawback to this work is one of adoption. The aws-cloudfront module is one of the oldest in the CDK, and is used by many customers.
+These changes won't break any of the existing customers, but all existing customers would need to rewrite their CloudFront constructs to take
+advantage of the functionality and ease-of-use benefits of the new L2s. For building an entirely new set of L2s to be worth the return on investment,
+the improvements to functionality and ergonomics needs to be substantial. Feedback is welcome on how to best capitalize on this opportunity and make
+the friendliest interface possible.
+
+# Rationale and Alternatives
+
+This RFC aims to take one of the older modules in the CDK and update it to the current set of design standards. By introducing secondary resource
+creation methods, factories for common L2-based origins, and flattening some of the top-level nested properties, we can offer an easier-to-use
+experience.
+
+The interface aims to make it easiest for the 90%+ use cases of having a single origin based on an S3 bucket (or other CDK construct), with a single
+default behavior, optionally slightly customized; it accomplishes this by exposing a single top-level `origin` and `behavior` and making us of
+factory methods to construct origins from buckets, load balancers, and other common origins. Beyond that straightforward use-case, the decision was
+made to represent the behaviors as members of origins, as each behavior must be associated with an origin. This introduces one area of cognitive
+complexity in terms of behavior ordering.
+
+Behaviors are ordered and precedence is used to determine how to route requests to origin(s). For example, origin #1 could have custom behaviors with
+path patterns of 'images/\*.jpg' and '\*.gif', and origin #2 could have a behavior on 'images/\*'. Depending on the ordering, a request for
+'images/foo.gif' may either be routed to origin #1 or #2. The approach taken ties behavior precedence to order of creation. An alternative would
+be to expose a flat list of behaviors, and allow the user to manipulate that list to change precedence. Ultimately, this was discarded as an overly-
+complex interface with diminishing benefits. However, feedback is welcome on a more elegant way to give users control of behavior ordering.
+
+# Adoption Strategy
+
+Once created, the new L2s can be used by existing CDK developers for new use cases, or by converting their existing CloudFrontWebDistribution usages
+to the new cloudfront.Distribution resource.
+
+# Unresolved questions
+
+1. What level of breaking changes are acceptable for the existing `IDistribution` and `CloudFrontWebDistribution` resources? Notably, the
```

Comment:
Have we dropped the approach of publishing this as a new module? If so - why? I feel like putting both new and old in the same module is going to confuse users.